### PR TITLE
improve the documentation and local experience of using these tools

### PR DESCRIPTION
**Pull request by vacas5**
## Jira Ticket

Before creating this pull request, make sure you have a separate ticket for the docs team to track their work. If you need assistance, ask in the #documentation Slack channel.

## Description of changes being made

Brought over the configuration for markdownlint and locally I integrated the VSCode plugins to make these linters easier to use. The package-lock changes are expected because I ran `npm i` with the updated node version, etc.

## Checklist

- [ ] Test all commands and procedures, if applicable.
- [ ] Create new PRs against `develop`, or `main` for hotfixes to production.
- [ ] Update all links if you are moving a page.
- [ ] Add release date to Release Notes page in the following format: <Package> was released on <Day>, <Month> <Year> Example: `Mesosphere® DC/OS™ 2.1.0 was released on 9, June 2020`

See the [contribution guidelines](./CONTRIBUTING.md) for more information.

### CODE REVIEW

File: .remarkrc.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+  "plugins": [
+    "remark-validate-links",
+    ["remark-lint-no-dead-urls", { "skipLocalhost": true }]
+  ]
+}
```
Comment (vacas5) [inline .remarkrc.json:6]: moving here so we can leverage in a future github action

File: README.md [hunk 2]
```diff
@@ -1,26 +1,29 @@
...
-## Development / Writing
+## Development
 
 Run `npm run dev` or `make build-development` from the root directory to spin up a live preview of the docs site based on your local changes. After about a minute the compilation will complete.
 
-You'll now be able to browse the docs at [http://localhost:3000/](http://localhost:3000/). Your browser will reflect any changes to pages in dkp almost immediately.
+You'll now be able to browse the docs at [http://localhost:3000/](http://localhost:3000/). Your browser will automatically reflect any changes to pages.
 
-`npm i` may need to be run from time to time if dependencies have changed, but this is rare.
+If dependencies have changed, which is rare but happens from time to time, run `npm i` again.
```
Comment (vacas5) [inline README.md:20]: you can see the impact of integrating Vale into VSCode :)

File: README.md [hunk 3]
```diff
@@ -95,4 +122,4 @@ Then anyone could run that pushed image and have a look if everything is alright
 docker run -p 5000:5000 -it mesosphere/archived_docs:next
 ```
 
-If you merge your branch to the `archive`-branch, jenkins will take care of building and pushing the image.
+If you merge your branch to the `archive`-branch, CI will take care of building and pushing the image.
```
Comment (vacas5) [inline README.md:125]: this whole section needs to be revised and removed. I think we'll just want to tag `main` to archive going forward and build a docker image from it. TBD.

File: package.json [hunk 4]
```diff
@@ -72,14 +72,8 @@
     "*.js": "eslint --cache --fix",
     "*.md": [
       "vale --config=./.vale.ini",
-      "remark --quiet --frail",
+      "remark --quiet --frail --no-stdout",
```
Comment (vacas5) [inline package.json:75]: prevents remark from spitting out all the markdown to the console.

**Review by clintonmedbery — APPROVED** [review]
Thanks for the improvements.

**Pull request merged**